portfolio/src/main/webapp/script.js

```diff
@@ -42,3 +42,505 @@ function visitLink(s){
+                }
+                //Rook moving patterns
+                else if((square.row - this.selected_square.row == 0 || 
+                    square.col - this.selected_square.col == 0) && (square.row != 
+                    this.selected_square.row || square.col != this.selected_square.col)){
+
+                    return true;
+                }
+            }
+            else if(selected_piece.type == 'king'){
+                if(Math.abs(square.row - this.selected_square.row) < 2 && 
+                    Math.abs(square.col - this.selected_square.col) < 2){
+                    
+                    selected_piece.moved = true;
+                    return true;
+                }
+
+                //Unimplemented castling feature
+                //TODO: Implement castling
+                else if(square.row == this.selected_square.row && 
+                    square.col == this.selected_col + 2){
+                    if(selected_piece.isWhite){
+                        if(!this.row[7][5].piece && !this.row[7][6].piece){
+                            selected_piece.moved = true;
+                            return 2;
+                        }
+                    }
+                    else{
+                        if(!this.row[0][5].piece && !this.row[0][6].piece){
+                            selected_piece.moved = true;
+                            return 3;
+                        }
+                    }
+                    return false;
+                }
+                else if(square.row == this.selected_square.row && 
+                    square.col == this.selected_col - 2){
+                    if(selected_piece.isWhite){
+                        if(!this.row[7][1].piece && !this.row[7][2].piece && !this.row[7][3].piece){
+                            selected_piece.moved = true;
+                            return 2;
+                        }
+                    }
+                    else{
+                        if(!this.row[0][1].piece && !this.row[0][2].piece && !this.row[0][3].piece){
+                            selected_piece.moved = true;
+                            return 3;
+                        }
+                    }
+                    return false;
+                }
+            }
+        }
+        return false;
+    }
+
+    squareClicked(square){
+
+        //if there is a previously selected square
+        if(this.selected_square){
+
+            //determine if move is legal
+            let legal = this.checkLegal(square, this.selected_square.piece);
+
+            //if the move is legal
+            if(legal == true){
+                //if there is a piece on the square
+                if(square.piece){
+                    //if the piece is the same color as the piece on the previously selected square
+                    if(square.piece.isWhite == this.selected_square.piece.isWhite){
+                        //TODO: make .moved false if the piece hasn't moved when reselecting
```

Comment:
Only make it false if it was false before it was selected. Or maybe you can have the setting of moved take place below where you are updating the image so it truly is when the piece moved.